app/src/heartbeat/heartbeat_controller_desktop.h

```diff
@@ -36,14 +36,52 @@ class HeartbeatController {
   // Asynchronously log a heartbeat, if needed
   void LogHeartbeat();
 
+  // Synchronously fetches and clears all heartbeats from storage and returns
+  // a JSON payload that has been compressed with gzip and base64 encoded.
+  // If there are no new heartbeats, an empty string is returned instead.
+  std::string GetAndResetStoredHeartbeats();
+
+  // Synchronously fetches and clears today's heartbeat from storage and returns
+  // a JSON payload that has been compressed with gzip and base64 encoded.
+  // If there is no new heartbeat, an empty string is returned instead.
+  std::string GetAndResetTodaysStoredHeartbeats();
+
  private:
+  friend class HeartbeatControllerDesktopTest_EncodeAndDecode_Test;
```

Comment:
You need to put these friend classes between this macro

```
#ifdef FIREBASE_TESTING
#end
```

Reply:
Done. Added this to the Cmake file for the test:
DEFINES
    -DFIREBASE_TEST